Firestore/Source/Core/FSTFirestoreClient.mm

```diff
@@ -393,13 +393,11 @@ - (void)getDocumentsFromLocalCache:(FIRQuery *)query
     ViewSnapshot snapshot = std::move(viewChange.snapshot).value();
     SnapshotMetadata metadata(snapshot.has_pending_writes(), snapshot.from_cache());
 
-    FIRQuerySnapshot *result = [[FIRQuerySnapshot alloc] initWithFirestore:query.firestore.wrapped
-                                                             originalQuery:query.query
-                                                                  snapshot:std::move(snapshot)
-                                                                  metadata:std::move(metadata)];
+    api::QuerySnapshot result(query.firestore(), query.query(), std::move(snapshot),
+                              std::move(metadata));
```

Comment:
Nit: I think `metadata` is a POD, in which case I'd rather not `move` it (I think it applies in more than one spot in this PR).

Reply:
I can see the argument going both ways, so ultimately I'll leave this to your judgment. One could argue that `move`ing PODs reduces clarity by making it appear that the class in question is more complicated than it actually is. `clang-tidy` [flags this](https://clang.llvm.org/extra/clang-tidy/checks/performance-move-const-arg.html), presumably on these grounds. On the other hand, it's a good point that `move`ing such types is more future-proof. Feel free to leave as is.